src/eureka/S1_detector_processing/ramp_fitting.py

```diff
@@ -87,6 +89,46 @@ def process(self, input):
             Updated for JWST version 1.3.3, code restructure
         '''
         with datamodels.RampModel(input) as input_model:
+
+            if self.s1_meta.update_sat_flags:
+                input_model = update_saturation.update_sat(input_model, 
+                                                           self.s1_log, 
+                                                           self.s1_meta)
+
+            if self.s1_meta.grouplevel_bg:
+                from ..S3_data_reduction import background as bkg
+                import astraeus.xarrayIO as xrio
```

Comment:
Imports should only be done at the top of the file, not in the middle of the code for flake8 compliance

Reply:
now in new function at top of file